BuildScript.Nuke/GenerateSbom/SolutionSbomGenerator.cs

```diff
@@ -116,25 +118,55 @@ public void Generate (Tool cycloneDxTool, IReadOnlyCollection<ProjectInfo> proje
 
       _outputFile.Parent.CreateDirectory();
 
-      File.Copy(_workingDirectory / cleanedSbomFileName, _outputFile, overwrite: true);
+      //The sbom cleaner only works with xml files, so we need to convert the sbom if we want a different format.
+      //Given a requested xml file, we can just copy
+      var outputSbomFileToCopy = _workingDirectory / cleanedSbomFileName;
+      if (_outputFile.Extension == ".xml")
+      {
+        File.Copy(outputSbomFileToCopy, _outputFile, overwrite: true);
+      }
+      else if (_outputFile.Extension == ".json")
+      {
+        DuplicateAsJsonToOutputFile(outputSbomFileToCopy, _outputFile);
+      }
 
       Log.Information($"Cleaned sbom has been copied to '{_outputFile}'.");
     }
   }
 
+  private void DuplicateAsJsonToOutputFile (AbsolutePath sbomFileToConvert, AbsolutePath finalJsonSbomLocation)
+  {
+    using var cleanedBomStream = File.OpenRead(sbomFileToConvert);
+    var finalBom = XMLSerializer.Deserialize(cleanedBomStream);
+
+    using var outputSbomJsonStream = File.Create(finalJsonSbomLocation);
+
+    JsonSerializer.SerializeAsync(finalBom, outputSbomJsonStream).GetAwaiter().GetResult();
+  }
+
   private void MergeSboms (string cleanedSbomFileName, string npmSbomFileName, AbsolutePath outputFile)
   {
     using var mainSbomFileStream = File.OpenRead(_workingDirectory / cleanedSbomFileName);
     using var npmSbomFileStream = File.OpenRead(_workingDirectory / npmSbomFileName);
 
-    var mainSbom = Serializer.Deserialize(mainSbomFileStream);
-    var npmSbom = Serializer.Deserialize(npmSbomFileStream);
+    var mainSbom = XMLSerializer.Deserialize(mainSbomFileStream);
+    var npmSbom = XMLSerializer.Deserialize(npmSbomFileStream);
+
+    var component = new Component { Type = Component.Classification.Library, Name = _solution.Name + "-root-component", Version = _version };
 
-    var combinedSbom = CycloneDXUtils.HierarchicalMerge([mainSbom, npmSbom], mainSbom.Metadata.Component);
+    var combinedSbom = CycloneDXUtils.FlatMerge([mainSbom, npmSbom], component);
 
     using var outputFileStream = File.Create(outputFile);
 
-    Serializer.Serialize(combinedSbom, outputFileStream);
+    if (outputFile.Extension.EndsWith("xml"))
+    {
+      XMLSerializer.Serialize(combinedSbom, outputFileStream);
+    }
+    if (outputFile.Extension.EndsWith("json"))
```

Comment:
As mentioned before I would prefer `outputFile.Extension == ".ext"` to remove the need to think about why we are using `.EndsWith(...)`

Reply:
Ah I only inspected it in the other case. thanks for finding this